### PR TITLE
first version of TikaContentExtractor

### PR DESCRIPTION
**Pull request by orhlulu**

### CODE REVIEW

File: src/main/java/innohack/gem/service/extract/TikaContentExtractor.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          results.getRecords().add(foundResults);
+        }
+
+        return results;
+      }
+    }
+    throw new IllegalStateException("No Tika Feature found");
+  }
+
+  private void populate(ExtractedRecords results, String matchedString, int rowIdx, String name)
```
Comment (onegang) [inline src/main/java/innohack/gem/service/extract/TikaContentExtractor.java:58]: Can remove these `populate` methods as they are not used?

File: src/main/java/innohack/gem/service/extract/TikaContentExtractor.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        String tikaContent = ((TikaFeature) feature).getContent();
+
+        Pattern pattern = Pattern.compile(regexString, Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(tikaContent);
+        int count = 0;
+
+        List<String> foundResults = Lists.newArrayList();
+        while (matcher.find()) {
+          count++;
+          String matchText = tikaContent.substring(matcher.start(), matcher.end());
```
Comment (onegang) [inline src/main/java/innohack/gem/service/extract/TikaContentExtractor.java:47]: Are we assuming no headers for this extraction? Or will it be an optional param specified by user? (if there's header, first row is header?)

File: src/main/java/innohack/gem/service/extract/TikaContentExtractor.java (hunk 1 above)
Comment (orhlulu) [reply src/main/java/innohack/gem/service/extract/TikaContentExtractor.java:58]: ok

File: src/main/java/innohack/gem/service/extract/TikaContentExtractor.java (hunk 2 above)
Comment (orhlulu) [reply src/main/java/innohack/gem/service/extract/TikaContentExtractor.java:47]: this one i also lost liao.. for tikaextraction is meant for pure textual and not only tabular content?

Comment (onegang) [reply src/main/java/innohack/gem/service/extract/TikaContentExtractor.java:47]: I also dunno... I am inferring from the UI... so we'll just assume extract blobs of text then. In future, can extend to extract blobs of text, table with headers and tables without headers (via more params)

**Pull request merged**